### ISSUE TITLE
Custom Blocked Access Page

### ISSUE THREAD

**Issue by hallberggroup** [AdguardTeam/AdGuardHome#454]
Would be great if there could be a settings page for changing the Blocked Page that shows. Like custom title and text and a possibillity to let the user to mail the admin.

**Comment by ameshkov** [comment]
Hm, I was almost sure that I've already created this task, but it appears I have not.

Thank you, we'll do it for sure!

**Comment by bzy39** [comment]
need safeBrowsingBlockHost and parentalBlockHost in config

**Comment by dany20mh** [comment]
This would be a good option if we can have

**Comment by rapdodge** [comment]
Any updates for this @ameshkov?

**Comment by ameshkov** [comment]
@rapdodge this one is not yet assigned to any milestone. We'll be reviewing all the popular feature requests after v0.95 update.

**Comment by Bluscream** [comment]
> rapdodge this one is not yet assigned to any milestone. We'll be reviewing all the popular feature requests after v0.95 update.

It is 0.96 already, can we get a milestone here?

**Comment by Bluscream** [comment]
Additionally i request a way to define a rule as being dropped
- In case of aghome filter `<rule>$drop` or in case of HOSTS filter `0.0.0.0 <rule>`

**Comment by Bluscream** [comment]
Here's a mockup for the block page:

![](https://user-images.githubusercontent.com/3318223/59064548-d02def80-88aa-11e9-903d-a7378bbb2fca.png)

**Comment by Eyeborgs** [comment]
It is also desirable that this feature be customizable to enable and disable it.

**Comment by ibksturm** [comment]
vote +1

**Comment by ameshkov** [comment]
@szolin 

For starters, let's make `safeBrowsingBlockHost` and `parentalBlockHost` configurable -- add them to the `yaml` configuration file.

**Comment by ameshkov** [comment]
Closed by https://github.com/AdguardTeam/AdGuardHome/commit/a9fbb93f0fd2fe9de2ad1857edeb9ff95e32b1f3

**Comment by ameshkov** [comment]
@jupiter09 you can change `parental_block_host ` and `safebrowsing_block_host` in the configuration (`AdGuardHome.yaml`) settings to a hostname where you host your block page version.

**Comment by tomcoleman** [comment]
> @jupiter09 you can change `parental_block_host ` and `safebrowsing_block_host` in the configuration (`AdGuardHome.yaml`) settings to a hostname where you host your block page version.

Hi im on version Version: 
v0.107.2

but when i check the adguardhome.yaml i do not see parental_block_host ` and `safebrowsing_block_host` ?

**Comment by EugeneOne1** [comment]
@tomcoleman, hello.  These settings are intended to be used only within browsing security and parental control web services.  You may achieve these pages by visiting some adult site.  In this case also make sure you have "Use AdGuard parental control web services" setting checked.  These requests are also shown in query log as **Blocked by Parental Control**.

The result of requesting domains blocked for another reason (e.g. filtering rules, blocked services, etc.) is configured by the `blocking_mode` setting.  See the [Configuration](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file) wiki page for all available values.

**Comment by outofsight** [comment]
Hello, I don't understand if currently there us an easy, AGH only, way to setup a block page, even not customized, to inform users a request was blocked by AGH.

**Comment by EugeneOne1** [comment]
@outofsight, it's unfortunately not possible.  The easiest way to achieve this currently described [here](https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#customblock).

**Comment by ajisetiawan716** [comment]
Saya telah membuat cuetom block page menggunakan github pages sebagai hostingnya, saya juga telah mengatur DNS agar terarahkan ke DNS github, DNS manajemen yang saya gunakan adalah cloudflare. Dan domainnya adalah family -block.dns.ajimedia.my.id. Saya telah mengaturnya sebagai host-nya untuk nilai dari parental_control_host, akan tetapi ketika saya membuka halaman yang sudah terfilter, mengapa saya diarahkan ke ip github bukan ke domain host yang telah diatur tadi? (Lihat pada gambar)

![Screenshot_2024-02-14-12-04-51-358_com android chrome-edit](https://github.com/AdguardTeam/AdGuardHome/assets/106381048/8f391141-f170-4569-84d7-b8e65793fa53)

**Comment by ajisetiawan716** [comment]
I have created a custom block page using github pages as the hosting, I have also set the DNS to point to the github DNS, the management DNS that I use is cloudflare.  And the domain is family -block.dns.ajimedia.my.id.  I have set it as the host for the value of parental_control_host, but when I open the filtered page, why am I redirected to the GitHub IP instead of the host domain that was set earlier?  (See picture)

![Screenshot_2024-02-14-12-04-51-358_com android chrome-edit](https://github.com/AdguardTeam/AdGuardHome/assets/106381048/590ce44a-9ff7-499e-a059-e57c9290975d)
